feat(QDOG-58): Create the add pet form page

arasantos opened the pull request:
- Create a page for adding a pet, matching Figma design as closely as possible. The difference is due to simplifying the responsive design code, as well as adding a reset button to clear the image.
- Add data that's used for new pet page to data folder to decouple UI and data
- Add validation methods for validating inputs to new pet form
- Add filehandling.ts to turn url into a file, so petImage can be initialized with a non-null value
- Add placeholder image
- Install mantine dates and dayjs for date picker in new-pet page
- Rename petOwner to pet-owner for consistent file name convention
- Add color used for button in new-pet page
- Format files

Desktop:
<img width="2541" height="1210" alt="image" src="https://github.com/user-attachments/assets/4aebd3e1-9ed7-47f9-8b37-c65cb00e7f81" />

Selecting image: (without toggle, but new one has a toggle)
<img width="2342" height="1046" alt="image" src="https://github.com/user-attachments/assets/1b36582c-8752-4f91-bbf7-7b5f4f9fd7ac" />
<img width="2529" height="1200" alt="image" src="https://github.com/user-attachments/assets/f6cbbc04-0d06-4bc7-8f3b-852e439f6760" />

Mobile: 
<img width="274" height="782" alt="image" src="https://github.com/user-attachments/assets/f6d95e76-d704-46f1-bbcc-3594647cf43a" />

Selecting image:
<img width="275" height="788" alt="image" src="https://github.com/user-attachments/assets/81fc6dba-774c-4fc9-98fe-132b5b22622a" />

Toggled on:
<img width="279" height="778" alt="image" src="https://github.com/user-attachments/assets/c89d211c-ed4c-478c-88da-ae7a5c41f579" />

arasantos committed:
feat(QDOG-58): Create add pet form page

Create a page for adding a pet, matching Figma design as closely as possible
Add data that's used for new pet page to data folder to decouple UI and data
Add validation methods for validating inputs to new pet form
Add placeholder image
Install mantine dates and dayjs for date picker in new-pet page
Rename petOwner to pet-owner for consistent file name convention
Add color used for button in new-pet page
Format files

arasantos committed:
feat(QDOG-58): Create add pet form page

Format files

arasantos committed:
feat(QDOG-58): Create add pet form page

Move validateStringValue and validateSelectedOption to general validation file
Format files

arasantos committed:
feat(QDOG-58): Create add pet form page

Rebase this branch and update some imports to follow the format in ts config]
Move all data to constants.ts

arasantos committed:
feat(QDOG-58): Create add pet form page

Make petImage initial value the placeholder image as file, instead of null

arasantos committed:
feat(QDOG-58): Create add pet form page

Export urlToFile and use in the new pet page

arasantos committed:
feat(QDOG-58): Create add pet form page

Change filename to match other file name convention

arasantos committed:
feat(QDOG-58): Create add pet form page

Removed a comment as it's not needed in that file

runavv commented on src/data/constants.ts:
```diff
@@ -16,3 +16,23 @@ export const provinces = [
 ];
+
+export const basicOptions: string[] = ["Yes", "No", "Unknown"];
+
```
Non blocking just thinking out loud: maybe we should have "Not Applicable" as an option? I was thinking for spayed/neutered it shouldn't be required but I then went here to see the options, unknown is good but I think you wouldn't spay/neuter like birds or lizards and so it wouldn't be unknown but also not yes/no

arasantos replied:
That's a great point! I added it to the options :)

runavv reviewed (approved):
Nice work! Love the references in code too

arasantos committed:
feat(QDOG-58): Create add pet form page

Add Not Applicable as an option for spayed/neutered

arasantos committed:
feat(QDOG-58): Create add pet form page

Add estimated toggle that redirects birthdate date picker value to the estimated date internally
Move validation for date to general validation file, and create separate for required and not required date

arasantos committed:
feat(QDOG-58): Create add pet form page

Change file name to follow name convention

arasantos commented:
Note: the commits added after the approvals were just from rebasing this branch to develop
